ff/src/fields/arithmetic.rs

```diff
@@ -155,19 +151,22 @@ macro_rules! impl_field_square_in_place {
             r[1] <<= 1;
 
             for i in 0..$limbs {
-                r[2 * i] = fa::mac_with_carry(r[2 * i], (self.0).0[i], (self.0).0[i], &mut carry);
-                r[2 * i + 1] = fa::adc(r[2 * i + 1], 0, &mut carry);
+                r[2 * i] = mac_with_carry!(r[2 * i], (self.0).0[i], (self.0).0[i], &mut carry);
+                #[allow(unused_assignments)]
```

Comment:
Where is the unused assignment coming from?

Reply:
The last value assigned to carry (inside the macro) is unused

Reply:
That makes sense, and we just assume that the compiler will remove that in an optimization pass. Could a comment be added to that affect?